### PR TITLE
Add grand potential phase field model example with strict mass conservation

### LINKED ISSUE #22159: Add example for grand potential phase field with strict mass conservation and improve documentation
## Reason
In the multi-phase grand potential phase-field model, slight variations is total system mass have been observed. These are believed to result from the fact that the evolution equation for concentration is transformed into an evolution equation for chemical potential, which introduces terms that can cause mass variation as a form of discretization error. The error tends to get worse as time steps increase in size, leading to the conclusion it is a form of time discretization error.

## Design
An alternative formulation has been derived that maintains a conservation equation for concentration, and introduces a relationship between composition and chemical potential that can be derived from the grand potential functional. An input file that implements this new formulation as an example problem will be added, along with improved documentation. This input file uses existing kernels and materials to implement the new formulations, and has been tested and shown to conserve mass to within solver tolerance level.

## Impact
Only adds additional example input file and documentation.

### PR DESCRIPTION
**Pull request by laagesen**
Also improves documentation. Closes #22159

### CODE REVIEW

**Comment by laagesen** [comment]
@sourabhkadambi could you review this once tests pass?

File: modules/phase_field/examples/multiphase/GrandPotential3Phase_masscons.i [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+  [./IC_w]
+    type = FunctionIC
+    variable = w
+    function = ic_func_w
+  [../]
+[]
+
+[Functions]
+  [./ic_func_w]
+    type = ParsedFunction
```
Comment (dschwen) [inline modules/phase_field/examples/multiphase/GrandPotential3Phase_masscons.i:75]:
```suggestion
    type = ConstantFunction
```

File: modules/phase_field/examples/multiphase/GrandPotential3Phase_masscons.i [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+    full = true
+  [../]
+[]
+
+[Postprocessors]
+  [./c_total]
+    type = ElementIntegralVariablePostprocessor
+    variable = c
+  [../]
+[../]
```
Comment (dschwen) [inline modules/phase_field/examples/multiphase/GrandPotential3Phase_masscons.i:329]: could you please hit format this input file

File: modules/phase_field/examples/multiphase/GrandPotential3Phase_masscons.i [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+
+[Executioner]
+  type = Transient
+  nl_max_its = 15
+  scheme = bdf2
+  solve_type = PJFNK
```
Comment (dschwen) [inline modules/phase_field/examples/multiphase/GrandPotential3Phase_masscons.i:335]: By the way this runs faster with `NEWTON`. You can then also drop the `[Preconditioning]` block as it gets added automatically for Newton solves.

**Comment by laagesen** [comment]
@lr272727 can you please review?

**Review by sourabhkadambi — DISMISSED** [review]

**Review by dschwen — APPROVED** [review]

**Pull request merged**